code/modules/mob/living/simple_animal/friendly/friendly_emote.dm

```diff
@@ -91,3 +91,12 @@
 	key = "purr"
 	key_third_person = "purrs"
 	message = "purrs."
+
+/datum/emote/living/sit/cat
```

Comment:
You can just make this `/datum/emote/living/simple_animal/pet/cat/sit`, and you won't need to update the `mob_type_allowed_typecache`. See above.